src/BikeSharing.DomainLogic/BikeWorkout.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using BikeSharing.DomainLogic;
+
+namespace Training
+{
+    public class Workout
+    {
+        public DateTime Date { get; set; }
+        public TimeSpan Duration { get; set; }
+        public double AverageHeartRate { get; set; }
+        public string Notes;
+
+        public Workout(DateTime date, TimeSpan duration, double averageHeartRate, string notes)
+        {
+            Date = date;
+            Duration = duration;
+            AverageHeartRate = averageHeartRate;
+            if (notes == null)
+            {
+                throw new ArgumentNullException();
+            }
+            Notes = notes;
+        }
+    }
+
+    public class DistanceWorkout : Workout
+    {
+        public double Distance { get; set; }
+        public double Pace { get; }
+        public DistanceWorkout(double distance, DateTime datetime, TimeSpan duration, double rate, string notes) : base(datetime, duration, rate, notes)
+        {
+            Distance = distance;
+            Pace = distance / duration.TotalHours;
```

Comment:
>TotalHours [](start = 39, length = 10)

who measures there time on hours? I want to know if I did a 5 minutes mile, not if I ran 20 miles per hour